compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -1870,7 +1870,8 @@ declare_lint! {
     ///
     /// ### Example
     ///
-    /// ```rust,compile_fail
+    #[cfg_attr(bootstrap, doc = "```ignore")]
+    #[cfg_attr(not(bootstrap), doc = "```rust,compile_fail")]
```

Comment:
Thanks to Boxy for helping me figure out how to actually do this. ✨

Reply:
Unfortunately we have multiple checks using hand-rolled parsers that don't understand this pattern, so I've had to “fix” one and trick the other.